### ISSUE TITLE
Autocomplete when writing % and then ; is broken

### ISSUE THREAD

**Issue by Sandstedt** [styled-components/vscode-styled-components#322]
**Describe the bug (including copyable syntax)**
When writing ex `max-width: 100%` IntelliSense starts auto-completing with a lot of nonsense, like none, 100%, auto or max-width. If I now write a semi-colon, `;`, this nonsense autocompleted word is attached to the end.

**Screenshot**

https://user-images.githubusercontent.com/5272401/131821247-08d2312b-3566-483a-b897-dc421d4e73e7.mp4

**To Reproduce**
* Inside a styled declaration, write ` max-width: 100%` and then write a semi-colon

**Expected behaviour**
Not having any autocomplete at all here. Other units, like `px` and `vh` works fine.

**Build environment (please complete the following information):**
- OS: Mac and Windows (same bug)
- VSCode Version: 1.59.1
- Extension Version 1.6.6

**Context**
This happen in an update sometime during the summer. Hasn't been a problem before.

**Comment by jasonwilliams** [comment]
Thanks for raising @Sandstedt

**Comment by jasonwilliams** [comment]
I may need to raise this with VSCode because I can't see anywhere the extension causing this behaviour. Its very odd and I still haven't got to the source of it.

**Comment by jasonwilliams** [comment]
Maybe @aeschli could be of some assistance here?

**Comment by jasonwilliams** [comment]
Raised: https://github.com/microsoft/vscode/issues/134328

**Comment by forsgren** [comment]
I'm having similar issues:

https://user-images.githubusercontent.com/164518/140289404-5b7cdb20-e872-419c-88f7-9d689c9b194a.mp4

**Comment by jasonwilliams** [comment]
If someone wants to fix this I would take a look at what the CSS Language server is doing. This may take a couple of windows so I would do as follows: (Using whatever windowing system your OS offers)

## Window 1

Follow https://github.com/microsoft/vscode-css-languageservice#development and get setup with that, make sure the breakpoint is working like in the readme

## Window 2

Follow https://github.com/styled-components/vscode-styled-components/blob/master/CONTRIBUTING.md#setting-up-for-development--debugging make sure the breakpoints are working here too. 

Then its a case of seeing why Window 1 works and Window 2 doesn't, it may be something different we sent to the CSS Language service, maybe the positioning or offset is slightly off.

**Comment by jasonwilliams** [comment]
If anybody wants to help fix this there’s a ticket open here https://github.com/microsoft/typescript-styled-plugin/issues/155

The full context (including reproduction steps) is here: https://github.com/microsoft/vscode/issues/134328#issuecomment-974200861

**Comment by jasonwilliams** [comment]
Fixed by: https://github.com/microsoft/typescript-styled-plugin/pull/156

**Comment by Sandstedt** [comment]
@jasonwilliams 

Great to hear! 🎉 Sorry didn't find/take any time to help out. Can I buy you a week's worth of coffee instead? 😅 (seriously, thank you so much!)

**Comment by karlhorky** [comment]
> Can I buy you a week's worth of coffee instead? 😅 (seriously, thank you so much!)

Yeah totally, seriously, do you have a donation page @jasonwilliams? Would also want to support you!

**Comment by jasonwilliams** [comment]
That’s very kind but I’m ok!
You can help by testing the latest release and letting me know if there’s any issues. It should be out now, see changelog and don’t forget to switch to the new version.

I will use prereleases in future for beta testing

**Comment by karlhorky** [comment]
Checked it out and reported back in https://github.com/styled-components/vscode-styled-components/issues/325#issuecomment-991872828 , thanks!
